extensions/mapstore/src/test/java/com/hazelcast/mapstore/mongodb/MongoGenericMapStoreTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.mapstore.mongodb;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.DataLinkConfig;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MapStoreConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.util.FilteringClassLoader;
+import com.hazelcast.jet.SimpleTestInClusterSupport;
+import com.hazelcast.jet.mongodb.datalink.MongoDataLink;
+import com.hazelcast.map.IMap;
+import com.hazelcast.mapstore.GenericMapStore;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.QuickTest;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.CreateCollectionOptions;
+import com.mongodb.client.model.ValidationOptions;
+import org.assertj.core.api.Assertions;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.example.Person;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.testcontainers.containers.MongoDBContainer;
+
+import java.util.ArrayList;
+
+import static com.hazelcast.jet.mongodb.impl.Mappers.defaultCodecRegistry;
+import static com.hazelcast.test.DockerTestUtil.assumeDockerEnabled;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.util.Lists.newArrayList;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class})
+public class MongoGenericMapStoreTest extends SimpleTestInClusterSupport {
```

Comment:
We already have 

https://github.com/hazelcast/hazelcast/pull/23837/files#diff-b6c45ec9e789fa2ed4f2e96363b4b128fd31fcf7a7ed327acdb3dce61a364354R134-R139

so maybe you could add a dependency on hazelcast-jet-mongodb test jar and re-use the `AbstractMongoDBTest`?

Reply:
I could in theory, but I use different hazelcast instance configuration here